<?xml version="1.0"?>
<launch>
	<arg name="robot_name" default="pioneer"/>
	<arg name="tracking_frame" default="$(arg robot_name)/base_link"/>
	<arg name="object_name" default="$(arg robot_name)/map"/>
	<arg name="velocity_topic" default="$(arg robot_name)/cmd_vel"/>
	<arg name="max_vel_factor" default="1.3"/>
	<arg name="ideal_distance" default="2"/>
	<arg name="time_delta" default="0.04"/>
	<arg name="distance_to_goal" default="0.35"/>
	<arg name="max_num_of_no_actor" default="75"/>
	<arg name="same_object_distance" default="0.3"/>
	<arg name="lower_relative_bound" default="0.4"/>
	<arg name="upper_relative_bound" default="2"/>

	<node name="relative_pose_publisher" type="relative_pose_publisher.py" pkg="tracking_navigation">
		<param name="tracking_frame" value="$(arg tracking_frame)"/>
		<param name="object_name" value="$(arg object_name)"/>
	</node>

	<include file="$(find tracking_navigation)/launch/move_base_pioneer.launch"/>
	<include file="$(find obstacle_detector)/launch/nodes.launch"/>

	<node name="velocity_adaptation_with_object_tracking" type="velocity_adaptation_with_object_tracking.py" pkg="tracking_navigation" output="screen">
		<param name="max_vel_factor" value="$(arg max_vel_factor)"/>
		<param name="ideal_distance" value="$(arg ideal_distance)"/>
		<param name="velocity_topic" value="$(arg velocity_topic)"/>
		<param name="time_delta" value="$(arg time_delta)"/>
		<param name="distance_to_goal" value="$(arg distance_to_goal)"/>
		<param name="max_num_of_no_actor" value="$(arg max_num_of_no_actor)"/>
		<param name="same_object_distance" value="$(arg same_object_distance)"/>
		<param name="lower_relative_bound" value="$(arg lower_relative_bound)"/>
		<param name="upper_relative_bound" value="$(arg upper_relative_bound)"/>
	</node>
</launch>
